pkg/pool-stable-phantom/contracts/StablePhantomPool.sol

```diff
@@ -908,9 +924,49 @@ contract StablePhantomPool is
 
     // Store the latest invariant based on the adjusted balances after the join or exit, using current rates.
     // Also cache the amp factor, so that the invariant is not affected by amp updates between joins and exits.
-    function _updateInvariantAfterJoinExit(uint256 currentAmp, uint256[] memory balancesWithoutBpt) internal {
+    function _updateInvariantAfterJoinExit(
+        uint256 currentAmp,
+        uint256[] memory balancesWithoutBpt,
+        uint256 preJoinExitInvariant,
+        uint256 virtualSupply,
+        uint256 bptAmount
+    ) internal {
+        uint256 postJoinExitInvariant = StableMath._calculateInvariant(currentAmp, balancesWithoutBpt);
+
+        // Compute the growth ratio between the pre- and post-join/exit balances.
+        // Note that the pre-join/exit invariant is *not* the invariant from the last join,
+        // but computed from the balances before this particular join/exit.
+
+        uint256 protocolSwapFeePercentage = getProtocolFeePercentageCache(ProtocolFeeType.SWAP);
+
+        if (protocolSwapFeePercentage > 0) {
+            uint256 invariantGrowthRatio = postJoinExitInvariant > preJoinExitInvariant
+                ? postJoinExitInvariant.divDown(preJoinExitInvariant)
+                : preJoinExitInvariant.divDown(postJoinExitInvariant);
+
+            // Compute the bptIncrease = (1 + bptAmount/virtualSupply)
+            uint256 bptIncreaseRatio = FixedPoint.ONE.add(bptAmount.divDown(virtualSupply));
```

Comment:
This assumes we're minting BPT

Reply:
Also, we'd have less error by simply doing `(bpt + supply) / supply`. It'd consider changing this function arguments to instead receive prejoinexit supply and post join exit supply instead (which will also let us determine if this is a join or an exit, if needed).